### DISCUSSION TITLE
Scheduling shows

### DISCUSSION THREAD

**Discussion by tomhaines** [savonet/liquidsoap#1581] [Q&A]
I am looking to program a radio station and I have two use cases that I'm curious if liquidsoap can handle.

I have read the docs and it seems promising but I haven't been able to confirm if I can do this.

In each case I have directories of shows and within each show directory is an ordered list of episode audio files for that particular show.

Use cases

1) Shows are scheduled on a day and time (ie, Show A plays on Monday at 8pm). Then each time that show's scheduled time occurs, it plays the next episode from a directory of episodes. Each day is the next episode. 

2) A list of all shows is randomized, and then the station goes through that list of randomized shows and plays one episode randomly selected from each show. That way you don't hear two episodes of a show until all shows have played. 

Curious to see if this is possible. Thank you.

**Comment by rastawarez** [comment]
1, 
I am thinking you can schedule a day of a week + time of day. You can create sources pointing to the playlist. 
The playlist can be a particular folder. monday= playlist(episode/monday)  and so on... 
You can use `playlist.once` to play the folder and stop

Something like that but I am not sure with syntax check doc.

```
path="/media/mp3"
monday=playlist.once("#{path}/episode/monday")
thursday=playlist.once("#{path}/episode/tuesday")

switch([
  ({ 1w and 8h }, monday),
  ({ 2w and 12h }, tuesday)
])

```
I am using queues and an external scheduler. Anyway you can achieve what do you want more than one way.

look here  http://mcfiredrill.github.io/blog/liquidsoap-scheduling)
